### ISSUE TITLE
Errors with Mixxx on a Mac (10.14.6)

### ISSUE THREAD

**Issue by thanks1music** [erikrichardlarson/unbox#44]
I'm getting Javascript errors when trying to run unbox with Mixxx. When I first select "Mixxx" I get this error:

<img width="419" alt="Screen Shot 2021-05-19 at 11 25 36 PM" src="https://user-images.githubusercontent.com/84488652/118914716-bf40e200-b8f9-11eb-92cd-cf5739283885.png">

If I click "OK" and "Continue", I get this error: 

<img width="414" alt="Screen Shot 2021-05-19 at 11 25 48 PM" src="https://user-images.githubusercontent.com/84488652/118914817-e0093780-b8f9-11eb-86d0-684f2b843ba5.png">

If it helps, on a 2019 15-inch Macbook Pro, running OSX 10.14.6. 

My Mixxx version is: 2.2.4 (build 2.2 r6798)

I would loooooove to get this working (especially if I can send the data to the computer that uses OBS). It will make ID-ing tracks during my online sets so much easier!

Thanks!

**Comment by erikrichardlarson** [comment]
Thanks for using the latest release! For the first error, this happens when the unbox app is in your downloads folder, if you move to your desktop or applications folder it should prevent this error, lmk if that doesn't work.

For the second error it seems that the artist or track data might be empty, do you think this is the case for some of your played tracks? I'll update the query to handle empty data.

**Comment by thanks1music** [comment]
OK, that seemed to do it. To be clear, theapp need to be directly in the Applications folder for me - it didn't work if I moved the whole unbox folder into applications. 

And you were correct about the track data - I was testing with a song that had no artist data in it. When I switched to a track with full data, it worked fine. Very cool! I initially had some trouble getting the text color to configure correctly - it was white on white, but I was able to get that working too. 

What in Mixxx determines "Now Playing", when they're are usually two tracks going (the active track and the cueing track)? From testing it out it seems to be based on track time? Also, will it cause any problems if I manually edit the HTML files to change the look & feel of the text. 

Last question - is it necessary to require a twitch login? I don't stream on twitch - I do have a login for viewing though (which I used). 

Thanks, this is really cool! I'll test it out with two computers later because (as you surmised) I use a separate computer for OBS.
